### ISSUE TITLE
Allow for built in webmentions.

### ISSUE THREAD

**Issue by rkingett** [ericgregorich/micro-blog-cards-theme#4]
I have yet to find a theme with this built in, so I thought I'd ask it here. Could you add in [webmentions](https://micro.blog/webmention) by default to this theme, perhaps?

[Since MicroBlog supports web mentions,](https://micro.blog/webmention) I've been trying to find a theme with webmentions already built in. I thought about making this a toggle, but I am honestly not sure how [these codes could be added with toggles.](https://micro.blog/webmention)

**Comment by rmdes** [comment]
Would love to have this also built in natively in the theme, as well as the other indieweb microformats https://www.amitgawande.com/2018/02/10/204300.html
for now I'm trying to adapt this tutorial and implement this with custom templates.

**Comment by ericgregorich** [comment]
I'll need to spend some time looking at what's involved with with adding web mentions. Maybe this weekend?

**Comment by rmdes** [comment]
If it helps I can share tomorrow a bunch of indieweb ready Hugo themes (most quite old) but that do have the indieweb parts implemented.

**Comment by ericgregorich** [comment]
That would be helpful. Thanks!EricOn Apr 27, 2023, at 6:42 PM, Ricardo ***@***.***> wrote:﻿
If it helps I can share tomorrow a bunch of indieweb ready Hugo themes (most quite old) but that do have the indieweb parts implemented.

**Comment by rmdes** [comment]
Step by step tutorial to "indiewebify" any hugo theme ; https://www.amitgawande.com/2018/02/10/204300.html

the most recent hugo theme that support indieweb principles appears to be https://github.com/CrowderSoup/soup
then neofeed theme https://github.com/victoriadrake/neofeed-theme
and finally the https://github.com/AngeloStavrow/indigo which supports both webmentions, indieauth and microformats2

**Comment by ericgregorich** [comment]
I've started working on this some. What I've finished is implemented on my own site for now and I'll push an update to the theme after more testing. 

My updates include additions particularly on the post page to support h-entry tags. I also added an h-card to my own site. That will be difficult to incorporate into the theme. It seems like a separate plugin could be better. 

Last, I believe Micro.blog and the Cards Theme already support Web-mentions through the conversations feature. Conversations need to enabled on your account for this to work. It validated successfully anyway.

**Comment by rmdes** [comment]
> My updates include additions particularly on the post page to support h-entry tags. I also added an h-card to my own site. That will be difficult to incorporate into the theme. It seems like a separate plugin could be better.

That's really nice of you, perhaps, as starting point for the h-card, we could update the readme with a paragraph about it and all things related indieweb?
- indicate which file to make the change
- provide the h-card code template

Regarding webmentions : have you tried it with the test tool : https://webmention.rocks/ ?

**Comment by ericgregorich** [comment]
> That's really nice of you, perhaps, as starting point for the h-card, we could update the readme with a paragraph about it and all things related indieweb?
Yes, I'm working on a blog post and updating the Readme. I just didn't have time to finish it yet.

I'll be looking into the webmentions further. I agree, it would be great to have full support.

**Comment by benborges** [comment]
Any updates on this, very interested to use this..

**Comment by ericgregorich** [comment]
Webmentions is already built into Micro.blog and the Cards Theme as described on this [webmentions](https://micro.blog/webmention) help page.

If you look at the source code of your site you should see a line that looks like this.

` <link rel="webmention" href="https://micro.blog/webmention" />`

and another line where it includes the conversation.js which displays the web methods on the page.

`<script type="text/javascript" src="https://micro.blog/conversation.js?url=....`

This line renders the responses on your page. Be sure the setting **Include conversation on post page** is enabled in your Design settings page.

![image](https://github.com/ericgregorich/micro-blog-cards-theme/assets/26981280/0ab1f797-619c-4663-83af-3fae4693add4)

Are you asking for something else regarding webmentions?
